### PR TITLE
refactor(storage): remove unnecessary ref to local_version_manager

### PR DESCRIPTION
**Pull request by zwang28**
## What's changed and what's your intention?

Lift `LocalVersionManager::pick_few_tables` to `SstableStore::sstables`, then in turn removing some unnecessary references in `LocalVersionManager`, `CompactorContext` and `SharedBufferUploader`.

### CODE REVIEW

File: src/bench/ss_bench/operations/write_batch.rs [hunk 1]
```diff
@@ -98,13 +98,6 @@ impl Operations {
             if let Some(context) = context {
                 if let Some(task) = self.meta_client.get_compact_task().await {
                     Compactor::compact(context.clone(), task).await;
-                    let last_pinned = context.local_version_manager.get_version().unwrap();
-                    let version = self
-                        .meta_client
-                        .pin_version(last_pinned.id())
-                        .await
-                        .unwrap();
-                    context.local_version_manager.try_set_version(version);
```
Comment (MrCroxx) [inline src/bench/ss_bench/operations/write_batch.rs:107]: Why pinning is removed here?

**Review by MrCroxx — COMMENTED** [review]
Rest LGTM.

File: src/bench/ss_bench/operations/write_batch.rs (hunk 1 above)
Comment (zwang28) [reply src/bench/ss_bench/operations/write_batch.rs:107]: Local versions is being updated periodically by another task in `HummockStorage`, so this code snippet is not required.

Comment (zwang28) [reply src/bench/ss_bench/operations/write_batch.rs]: This code snippet originally attempts to ensure the version after compaction is available locally during benchmark.
It doesn't use pin_version properly but can be a temporary workaround for now.
I've resumed it and added some comments.

**Review by MrCroxx — APPROVED** [review]
LGTM

**Pull request merged**